### ISSUE TITLE
Random disconnections

### ISSUE THREAD

**Issue by mimmus68** [h4/LYWSD02-home-assistant#7]
I found another problem that after a while the sensor works correctly, it loses the connection and I have to restart hassio to make it work again.
I also found the same problem with the round sensor and solved it with a custom component
https://community.home-assistant.io/t/xiaomi-mijia-bluetooth-temperature-humidity-sensor-compatibility/43568/147

There are possible solutions

thanks

I attach a custom component file

[mi_bt.zip](https://github.com/h4/LYWSD02-home-assistant/files/3774964/mi_bt.zip)

**Comment by h4** [comment]
Thanks! I'll hope this sample helps me to fix my component.

**Comment by h4** [comment]
So, I looked at this component, it uses BLE-advertising instead of subscription (which my component uses). Pros: it may be more stable, Cons: it can't deal with extended data (like time, units, history).

Thanks again for help. I am hope, I can fix disconnect errors soon.

**Comment by clau-bucur** [comment]
Have you switched to using BLE-advertising?

**Comment by h4** [comment]
@clau-bucur no, but if you looking for it – please see https://github.com/h4/LYWSD02-home-assistant/issues/12, there are an link to alternative component, which supports it.

**Comment by clau-bucur** [comment]
Thank you! I've installed that and it's working nicely.
